db/models/orders.js

```diff
@@ -0,0 +1,141 @@
+// grab our db client connection to use with our adapters
+const client = require("../client");
+
+const getAllOrders = async () => {
+  try {
+    const { rows: orders } = await client.query(
+        `
+            SELECT * FROM orders;
+        `
+    );
+    return orders;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getOrderById = async (id) => {
+  try {
+    const {
+      rows: [order],
+    } = await client.query(
+        `
+            SELECT * FROM orders
+            WHERE id = $1;
+        `,
+      [id]
+    );
+    return order;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getOrderByUser = async ({ username }) => {
+  try {
+    const {
+      rows: [order],
+    } = await client.query(
+        `
+            SELECT orders.*, users.username AS "customerName"
+            FROM orders
+            JOIN users ON orders."userId" = users.id
+            WHERE username = $1;
+        `,
+      [username]
+    );
+    return order;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getOrderByStatus = async ({ status }) => {
+  try {
+    const {
+      rows: [order],
+    } = await client.query(
+        `
+            SELECT * FROM orders
+            WHERE status = $1
+        `,
+      [status]
+    );
+    return order;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const createOrder = async ({ userId, address, status }) => {
```

Comment:
I feel like something's not right here with userId but I'm not sure how to approach it. technically when a customer is placing an order, they wouldn't be entering their userId but rather their name... right?🤔